### DISCUSSION TITLE
please help me

### DISCUSSION THREAD

**Discussion by Seungzu** [hyperledger/fabric-chaincode-node#430] [General]
# 1. Build a network passively
- I am now trying to build test-network manually by entering commands directly without using network.sh.
```sh
# These two lines are commands provided in fabric docs.
# ./network.sh will not be used.
$ ./network.sh up createChannel -c mychannel -ca
$ ./network.sh deployCC -ccn basic -ccp ../asset-transfer-basic/chaincode-javascript/ -ccl javascript
```

# 2. Sequence of operations
- This is the task sequence for manual construction of test-network.
- Refer to the network.sh file below.
```sh
# ca container - orderer, org1, org2
$ docker-compose -f compose/compose-ca.yaml -f compose/docker/docker-compose-ca.yaml up -d

# Setting environment variables
export PATH=${PWD}/../bin:$PATH
export FABRIC_CFG_PATH=$PWD/../config/

# Generate certificate (orderer, org1, org2)
. organizations/fabric-ca/registerEnroll.sh
createOrg1
createOrg2
createOrderer

# Create connection.json file
./organizations/ccp-generate.sh
```
It worked fine up to this point.

# 3. next step & error
- The next task requires creating orderer, org1, and org2 containers.
- I think the command to create orderer, org1, and org2 containers is one of the two below.
- However, errors occur in both cases.
```sh
docker-compose -f compose/compose-test-net.yaml -f compose/docker/docker-compose-test-net.yaml up -d
# An error occurs as follows
# invalid spec: :/host/var/run/docker.sock: empty section between colons

docker-compose -f compose/compose-test-net.yaml up -d
# The orderer container was created normally.
# The org1 and org2 containers are output as “exited”.
# The logs of org1 and org2 containers are as follows.
# Fatal error when initializing core config: FABRIC_CFG_PATH /etc/hyperledger/peercfg does not exist
```

# 4. Please save me
- If you know a solution, please let me know.

**Comment by bestbeforetoday** [comment]
Looking at [docker-compose-test-net.yaml](https://github.com/hyperledger/fabric-samples/blob/ce5aa883eed78bb064edc7e9eabec99d9f05269e/test-network/compose/docker/docker-compose-test-net.yaml), it is performing a [volume mount using a DOCKER_SOCK environment variable](https://github.com/hyperledger/fabric-samples/blob/ce5aa883eed78bb064edc7e9eabec99d9f05269e/test-network/compose/docker/docker-compose-test-net.yaml#L19). I guess you do not have this set so the volume mount spec is invalid. Notice that the `network.sh` script (normally used to drive this network set up) sets this environment variable, with the default value being [`/var/run/docker.sock`](https://github.com/hyperledger/fabric-samples/blob/ce5aa883eed78bb064edc7e9eabec99d9f05269e/test-network/network.sh#L479-L481)
